python3Packages.ffmpy: unbreak build with uv-build >=0.8.0

bengsparks opened the pull request:
Ensure `python3Packages.ffmpy` continues to build after #426319 is merged.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and others READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bengsparks committed:
python3Packages.ffmpy: unbreak build with uv-build >=0.8.0

bengsparks committed:
python3Packages.ffmpy: move uv-build to build-system

bengsparks commented on pkgs/development/python-modules/ffmpy/default.nix:
```diff
@@ -24,22 +24,28 @@ buildPythonPackage rec {
+    # uv-build in nixpkgs is now at 0.8.0, which otherwise breaks the constraint set by the package.
+    + ''
+      substituteInPlace pyproject.toml \
+        --replace-fail 'requires = ["uv_build>=0.7.9,<0.8.0"]' 'requires = ["uv_build>=0.8.0,<0.9.0"]'
```
Excellent point 😄 I can get to it tomorrow evening

GaetanLepage commented on pkgs/development/python-modules/ffmpy/default.nix:
```diff
@@ -24,18 +24,24 @@ buildPythonPackage rec {
-        'executable: str = "${ffmpeg-headless}/bin/ffmpeg",'
+  postPatch =
+    ''
+      # default to store ffmpeg
```
Ideally, comments should be kept outside of nix strings.

GaetanLepage commented on pkgs/development/python-modules/ffmpy/default.nix:
```diff
@@ -24,22 +24,28 @@ buildPythonPackage rec {
+      substituteInPlace ffmpy/ffmpy.py \
+        --replace-fail \
+          'executable: str = "ffmpeg",' \
+          'executable: str = "${ffmpeg-headless}/bin/ffmpeg",'
```
```suggestion
          'executable: str = "${lib.getExe ffmpeg-headless}",'
```
nit

GaetanLepage reviewed (commented):
Overall LGTM

bengsparks committed:
python3Packages.ffmpy: refactor

Prefer lib.getExe over hardcoding the ffmpeg binary path,
Move comments out of patches

GaetanLepage reviewed (approved):
Oh I didn't get that this could get merged first. All good then!

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 426347`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16442617168


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.ffmpy</li>
    <li>python312Packages.ffmpy.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.ffmpy</li>
    <li>python313Packages.ffmpy.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>zotify</li>
    <li>zotify.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.ffmpy</li>
    <li>python312Packages.ffmpy.dist</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python313Packages.ffmpy</li>
    <li>python313Packages.ffmpy.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>zotify</li>
    <li>zotify.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 22 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.ffmpy</li>
    <li>python312Packages.ffmpy.dist</li>
    <li>python313Packages.ffmpy</li>
    <li>python313Packages.ffmpy.dist</li>
    <li>zotify</li>
    <li>zotify.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 22 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python312Packages.ffmpy</li>
    <li>python312Packages.ffmpy.dist</li>
    <li>python313Packages.ffmpy</li>
    <li>python313Packages.ffmpy.dist</li>
    <li>zotify</li>
    <li>zotify.dist</li>
  </ul>
</details>
